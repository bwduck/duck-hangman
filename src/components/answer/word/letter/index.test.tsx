import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AnswerLetter from './index';

describe('<AnswerLetter/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<AnswerLetter show letter="A" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the letter properly', () => {
    const wrapper = shallow(<AnswerLetter show letter="A" />);
    const span = wrapper.find('span');
    expect(span.text()).toBe('A');
  });

  it('does not render letter if show is false', () => {
    const wrapper = shallow(<AnswerLetter show={false} letter="A" />);
    const span = wrapper.find('span');
    expect(span.text()).toBe('');
  });
});
