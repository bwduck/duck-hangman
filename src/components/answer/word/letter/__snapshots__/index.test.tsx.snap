// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnswerLetter/> renders and matches the snapshot 1`] = `
<span
  className="c-answer-letter"
>
  A
</span>
`;
